modules/@apostrophecms/attachment/index.js

```diff
@@ -707,7 +734,10 @@ module.exports = {
         const batchSize = 100;
         let lastId = '';
         while (true) {
-          const docs = await self.db.find({ _id: { $gt: lastId } }).limit(batchSize).sort({ _id: 1 }).toArray();
+          const docs = await self.db.find({
+            ...(criteria || {}),
```

Comment:
A bug that got in the way: the `each` method for attachment migrations was ignoring its `criteria` argument.